examples/api/lib/material/text_form_field/text_form_field.2.dart

```diff
@@ -0,0 +1,143 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+/// Flutter code sample for [TextFormField].
+
+void main() => runApp(const TextFormFieldExampleApp());
+
+class TextFormFieldExampleApp extends StatelessWidget {
+  const TextFormFieldExampleApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return const MaterialApp(
+      home: TextFormFieldExample(),
+    );
+  }
+}
+
+class TextFormFieldExample extends StatefulWidget {
+  const TextFormFieldExample({super.key});
+
+  @override
+  State<TextFormFieldExample> createState() => _TextFormFieldExampleState();
+}
+
+class _TextFormFieldExampleState extends State<TextFormFieldExample> {
+  late final TextEditingController controller;
+  final GlobalKey<FormState> formKey = GlobalKey<FormState>();
+  String? forcedErrorText;
+  bool isLoading = false;
+  // username as a key, error text as value.
+  final Map<String, String> cachedServerErrors = <String, String>{};
+
+  @override
+  void initState() {
+    super.initState();
+    controller = TextEditingController();
+  }
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Material(
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 24),
+        child: Center(
+          child: Form(
+            key: formKey,
+            child: Column(
+              mainAxisAlignment: MainAxisAlignment.center,
+              children: <Widget>[
+                TextFormField(
+                  forceErrorText: forcedErrorText,
```

Comment:
Elsewhere I recommended getting rid of `forcedErrorText`, but if you keep it, I would drop the "d" so that its name matches the paramter `forceErrorText`.